dev-tools/validation/test_design_capability_allocation.py

```
@@ -0,0 +1,256 @@
+    lines = path.read_text(encoding="utf-8").splitlines()
+    matching_lines = [index for index, line in enumerate(lines) if marker in line]
+    if len(matching_lines) != 1:
+        raise AssertionError(f"expected exactly one line containing {marker!r} in {path}")
+    index = matching_lines[0]
+    if lines[index].count(old) != 1:
+        raise AssertionError(f"expected exactly one mutation target in {path}: {old!r}")
+    lines[index] = lines[index].replace(old, new, 1)
+    path.write_text("\n".join(lines) + "\n", encoding="utf-8")
+
+
+def main() -> None:
```

Ruff (0.15.21)

[warning] Too many statements (65 > 50)

(PLR0915)